modules/contact-form/grunion-contact-form.php

```diff
@@ -1380,18 +1395,24 @@ function __construct( $attributes, $content = null ) {
 	 *	- used to receate shortcode when user uses do_shortcode
 	 *
 	 * @param string $content
+	 * @param array $attributes
+	 * @param string $hash
 	 */
-	static function store_shortcode( $content = null, $attributes = null ) {
+	static function store_shortcode( $content = null, $attributes = null, $hash = null ) {
 
 		if ( $content != null and isset( $attributes['id'] ) ) {
 
-			$shortcode_meta = get_post_meta( $attributes['id'], '_g_feedback_shortcode', true );
+			if ( empty( $hash ) ) {
+				$hash = sha1( json_encode( $attributes ) . $content );
+			}
+
+			$shortcode_meta = get_post_meta( $attributes['id'], "_g_feedback_shortcode_{$hash}", true );
 
 			if ( $shortcode_meta != '' or $shortcode_meta != $content ) {
-				update_post_meta( $attributes['id'], '_g_feedback_shortcode', $content );
+				update_post_meta( $attributes['id'], "_g_feedback_shortcode_{$hash}", $content );
```

Comment:
We should add something that deletes all `_g_feedback_shortcode_*` postmeta on post update and then regenerates them for all new versions of the shortcodes or something -- or maybe only if they've changed.  Unsure, but I don't want to leave a bunch of old chaff cluttering the db if users keep changing forms.